Danki2/Assets/Scripts/Abilities/InstantCast/Roll.cs

```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+
+public class Roll : InstantCast
+{
+    public Roll(AbilityContext context) : base(context)
+    {
+    }
+
+    public override void Cast()
+    {
+        Actor caster = Context.Owner;
+
+        Vector3 targetPosition = Context.TargetPosition - caster.transform.position;
+        targetPosition.y = caster.transform.position.y;
+
+        caster.LockMovement(
+            0.15f,
+            caster.GetStat(Stat.Speed) * 4,
```

Comment:
I'd pull these two number out into private static readonly fields, something like `rollDuration` and `rollSpeedMultiplier`. I think ideally we'd be able to configure this kind of stuff through code somehow (so you could increase your roll speed somehow for example). But we can certainly kick that down the road.

Reply:
Yeah good point - it makes me think we could easily have tiers for each ability, in most cases just changing some numbers, and in other cases adding effects or casting an extra fireball or something